network_node.go

```diff
@@ -21,11 +22,12 @@ import (
 )
 
 type NetworkNode struct {
-	Identity peer.ID // the local node's identity
-	Kad      *kad.IpfsDHT
-	PeerHost host.Host // the network host (server+client)
-	Network  *BasicVideoNetwork
-	streams  map[peer.ID]*BasicStream
+	Identity       peer.ID // the local node's identity
+	Kad            *kad.IpfsDHT
+	PeerHost       host.Host // the network host (server+client)
+	Network        *BasicVideoNetwork
```

Comment:
Re-name 'streams' to 'outStreams', add a RwLock so we can protect it from race conditions.